src/backend/joanie/core/models/products.py

```diff
@@ -272,6 +273,7 @@ def save(self, *args, **kwargs):
 
             self.validate()
 
+    @transaction.atomic
     def validate(self):
```

Comment:
Looks like a bad name considering what the method is doing?

Reply:
What do you think of `enroll_to_single_course_runs`